server/controllers/threads.controller.js

```diff
@@ -70,22 +90,41 @@ function create(req, res, next) {
           ThreadId: thread.id,
           LastMessageId: thread.id //adding this while creating is okay as this is the first message in thread
         }).then((message) => {
+
+          //Get lastMessageId added to Thread
+          Thread.update({
+            lastMessageId: message.id
```

Comment:
Yes. This might be a better way to handle the One-to-One r/ship between a thread and it's last message. If this works we should perhaps get rid of the `Thread.hasOne(Message, {as: 'LastMessage'});` directive and have the foreign key on the Thread as opposed to the Message while simply updating the thread every time a new message is created as is done here.